beacon/engine/types.go

```diff
@@ -14,7 +14,7 @@
 // You should have received a copy of the GNU Lesser General Public License
 // along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
 
-package beacon
```

Comment:
Moving `core/beacon` to `beacon/engine` kinda warrants the question whether we also want to move `eth/catalyst/` to something more descriptive, maybe `beacon/eth` or `eth/engine` (same with les/catalyst)